### PR TITLE
Added catch to unhandled promises rejections

### PR DESCRIPTION
**Pull request by SimonBackx**
refs https://github.com/TryGhost/Team/issues/2370

- Reduces amount of unhandled promise rejections

### CODE REVIEW

File: ghost/core/core/server/services/mega/mega.js [hunk 1]
```diff
@@ -571,19 +571,25 @@ async function createEmailBatches({emailModel, memberRows, memberSegment, option
...
     if (emailRetried) {
-        pendingEmailHandler(emailModel, options);
+        await pendingEmailHandler(emailModel, options);
     }
 };
 
 function listen() {
-    events.on('email.added', pendingEmailHandler);
-    events.on('email.edited', statusChangedHandler);
+    events.on('email.added', (emailModel, options) => pendingEmailHandler(emailModel, options).catch((e) => {
```
Comment (naz) [inline ghost/core/core/server/services/mega/mega.js:585]: This is good imo 👍

File: ghost/core/core/server/services/email-analytics/jobs/fetch-latest/index.js [hunk 2]
```diff
@@ -47,4 +47,7 @@ if (parentPort) {
             process.exit(0);
         }, 1000);
     }
-})();
+})().catch((e) => {
+    console.error(e); // eslint-disable-line no-console
```
Comment (naz) [inline ghost/core/core/server/services/email-analytics/jobs/fetch-latest/index.js:51]: The errors thrown in jobs are handled [by this code](https://github.com/TryGhost/Ghost/blob/520b19e31376778a068165c34dac12cd8d6ae988/ghost/core/core/server/services/jobs/job-service.js#L13-L15). 

Error handling is part of the bree instance options - https://github.com/breejs/bree/tree/4f388fe70c1c27293ecff0bd6c8f8bd8a649405b#instance-options (afair we added this feature 😊 )

Comment (SimonBackx) [reply ghost/core/core/server/services/email-analytics/jobs/fetch-latest/index.js:51]: I'll remove these

Comment (SimonBackx) [reply ghost/core/core/server/services/email-analytics/jobs/fetch-latest/index.js:51]: Done! ✅

**Review by naz — APPROVED** [review]
yup, these make sense :+1:

**Pull request merged**